src/components/Cart/Address/add-new-address.jsx

```diff
@@ -0,0 +1,246 @@
+import { useState } from "react";
+import { useAuthContext, useData } from "../../../context";
+import { updateAddress } from "../../../services";
+
+const defaultData = {
+    name: "",
+    houseNo: "",
+    street: "",
+    locality: "",
+    city: "",
+    district: "",
+    state: "",
+    country: "India",
+    zipCode: "",
+    mobileNo: "",
+};
+
+const AddNewAddress = ({ isEditable, setIsEditable, IsAdd, setIsAdd, editAddress = defaultData }) => {
+    const { dispatch, setIsError } = useData();
+    const [address, setAddress] = useState(editAddress);
+    const [errorMsg, setErrorMsg] = useState({})
+    const { setShowLoader } = useAuthContext();
+
+    const validationHandler = () => {
+        let errorStatus = true;
+        if (!/^[6-9][0-9]{9}$/.test(address.mobileNo)) {
+            setErrorMsg((msg) => ({ ...msg, mobileNo: "Please Enter a Valid Mobile No" }));
+            errorStatus = false;
+        }
+        if (!/^[1-9][0-9]{5}$/.test(address.zipCode)) {
+            setErrorMsg((msg) => ({ ...msg, zipCode: "Please Enter a Valid Zip Code" }));
+            errorStatus = false;
+        }
+
+        return errorStatus
+    }
```

Comment:
- this is a type of common method you can move these type of functions in other helper files to make code more clean.

Reply:
Yes as a refactoring I will do this one.